src/utils/layout.ts

```diff
@@ -0,0 +1,31 @@
+import type { Layout, LayoutProperties } from "./types";
+
+import { useMediaQuery } from "react-responsive";
+
+const DEFAULT_LAYOUT = { isMobile: true };
+const DEFAULT_LAYOUT_KEY = 'isMobile';
+
+/**
+ * Detect current device
+ */
+export const useAppLayout = (): Layout => {
+  const isDesktopOrLaptop = useMediaQuery({ minWidth: 992 });
```

Comment:
Is it possible that there are >1 layout types are `true`. If not, I don't see the reason to return `true false false` or `false true false` or `false false true` map. It is excess data.

Reply:
Yes I see, I have simplified logic.